### DISCUSSION TITLE
[Flutter] null value in auto-increment id column on insert

### DISCUSSION THREAD

**Discussion by brericha** [supabase/supabase#13106] [Questions] [answered]
In one of my tables I have an int8 column as its primary key with the "is identity" flag enabled. When I try to insert a new row I'm getting an error stating `null value in column "id" of relation "tasks" violates not-null constraint`. My understanding is that the id column would be assigned by the database upon insert. Am I missing a step or something?

**Comment by GaryAustin1** [comment, marked as answer]
Are you providing the column name with null in your insert?   Insert values will override even is identity. You need to leave the column out totally.

**Reply by brericha** [reply]
That was exactly it! I'm using json_serializable and it's toJson function was passing the null value, thanks!
